addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -677,6 +679,85 @@ private boolean performImageTagAttack(HtmlContext context, HttpMessage msg, Stri
         return false;
     }
 
+    private boolean performAttributeAttack(HtmlContext context, HttpMessage msg, String param) {
+        String attackString = ACCESSKEY_ATTRIBUTE_ALERT;
+        List<HtmlContext> context2 = performAttack(msg, param, attackString, context, 0);
+        if (context2 == null) {
+            return false;
+        }
+        for (HtmlContext ctx : context2) {
+            Map<String, String> attrMap = ctx.getTagAttributes();
+            if (attrMap.containsKey("accesskey")
+                    && attrMap.get("accesskey").equals("x")
```

Comment:
Worth adding something like `HtmlContext#hasAttribute(String name, String value)` to avoid repeating this in all places.